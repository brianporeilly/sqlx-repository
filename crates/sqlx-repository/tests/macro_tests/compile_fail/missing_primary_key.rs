//! Test that repository derive fails when no primary key is specified

use sqlx_repository::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow, Repository)]
#[repository(table = "users")]
pub struct User {
    pub name: String,  // No id field and no #[repository(primary_key)] attribute
    pub email: String,
}

fn main() {}