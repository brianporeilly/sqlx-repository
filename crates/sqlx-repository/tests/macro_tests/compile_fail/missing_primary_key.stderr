error: Repository structs must have a primary key field named 'id: i32'.

       Example:
       #[derive(Repository)]
       #[repository(table = "users")]
       pub struct User {
           pub id: i32,  // ← Add this field
           pub name: String,
           pub email: String,
       }
  --> tests/macro_tests/compile_fail/missing_primary_key.rs:7:1
   |
7  | / #[repository(table = "users")]
8  | | pub struct User {
9  | |     pub name: String,  // No id field and no #[repository(primary_key)] attribute
10 | |     pub email: String,
11 | | }
   | |_^
